pkg/connection/settings.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2022 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connection
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+// Flags defines Flags that may be bound to the Settings. Use like:
+// `cmd.PersistentFlags().AddFlagSet(connection.Flags)`
+var Flags *pflag.FlagSet
+
+var configPath string
+var activeConfigPointerFilename = "active_config"
+
+func init() {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		panic(err)
```

Comment:
nit: I think there's something in the style guide about panics, but it seems pretty hard to get an error here: https://cs.opensource.google/go/go/+/refs/tags/go1.18.4:src/os/file.go;l=500